operator_cxx/contrib/quantization_int8-inl.h

```diff
@@ -0,0 +1,505 @@
+    }
+  }
+};
+
+void find_max(const OpContext &ctx, const TBlob &data, mshadow::Stream<xpu> *s, 
+              mshadow::Tensor<xpu, 1, char> &temp_reduce_space, TBlob &in_min_t, TBlob &in_max_t,
+              const mxnet::TShape &src_shape, const mxnet::TShape &dst_shape){
+    using namespace mshadow;
+    using namespace mshadow::expr;
+    broadcast::Reduce<red::minimum, 2, real_t, mshadow::op::identity>(
+        s, in_min_t.reshape(dst_shape), kWriteTo, temp_reduce_space, data.reshape(src_shape));
+    broadcast::Reduce<red::maximum, 2, real_t, mshadow::op::identity>(
+        s, in_max_t.reshape(dst_shape), kWriteTo, temp_reduce_space, data.reshape(src_shape));
+
+    // the maxabs value is save in in_max_t
+    mxnet_op::Kernel<find_maxabs, xpu>::Launch(s, 1, in_min_t.dptr<real_t>(), in_max_t.dptr<real_t>());
+}
+
+  virtual void Forward(const OpContext &ctx,
+                       const std::vector<TBlob> &in_data,
+                       const std::vector<OpReqType> &req,
+                       const std::vector<TBlob> &out_data,
+                       const std::vector<TBlob> &aux_states) {
+    using namespace mshadow;
+    using namespace mshadow::expr;
+    CHECK_EQ(in_data.size(), 1U);
+    CHECK_EQ(out_data.size(), 1U);
+    CHECK_EQ(req.size(), 1U);
+    CHECK_EQ(req[Quantization_int8_enum::kOut], kWriteTo);
+
+    Stream<xpu> *s = ctx.get_stream<xpu>();
+    Tensor<xpu, 4, DType> data;
+    Tensor<xpu, 4, DType> out;
+    if (in_data[Quantization_int8_enum::kData].ndim() == 2) {
+      Shape<4> dshape = Shape4(in_data[Quantization_int8_enum::kData].shape_[0],
+                               in_data[Quantization_int8_enum::kData].shape_[1], 1, 1);
+      data = in_data[Quantization_int8_enum::kData].get_with_shape<xpu, 4, DType>(dshape, s);
+      out = out_data[Quantization_int8_enum::kOut].get_with_shape<xpu, 4, DType>(dshape, s);
+    } else {
+      data = in_data[Quantization_int8_enum::kData].get<xpu, 4, DType>(s);
+      out = out_data[Quantization_int8_enum::kOut].get<xpu, 4, DType>(s);
+    }
+    Tensor<xpu, 1, DType> aux = aux_states[Quantization_int8_enum::kMinmax].get<xpu, 1, DType>(s);
+
+    /*
+    for weight per channel, OIHW, the axis should be the O dimension
+    for all per channel, OIHW the axis should be the I dimension, currently this is not considered
+    transpose conv weight is also not considered
+    */
+    // print_host(data.dptr_, data.shape_.Size(), "in_data");
+    if (ctx.is_train > 0 && quant_countdown > 0) {
+      mshadow::Copy(out, data, s);
+      // Assign(out, req[Quantization_int8_enum::kOut], data);
+      quant_countdown = quant_countdown - 1;
+    }
+    else {
+      if (param_.quant_mode == std::string("minmax")) {
+        // if (param_.quant_mode == 0) {
+        // Assign(out, req[Quantization_int8_enum::kOut], data);
+        // mshadow::Copy(out, data, s);  
+        uint64_t WORKSPACE_LIMIT = 1024 * 1024; // allocate 1MB
```

Comment:
Hardcoding in cpp is NOT acceptable since fix this hardcoding requires time-consuming re-compilation.